mmv1/products/discoveryengine/ChatEngine.yaml

```diff
@@ -0,0 +1,175 @@
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'industryVertical'
+    description: |
+      Industry Vertical
+    immutable: true
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      displayName
+    required: true
+  - !ruby/object:Api::Type::Array
+    name: 'dataStoreIds'
+    min_size: 1
+    description: |
+      Data Store IDs, Data Stores cannot be removed once added.
+      Removing a Data Store from Chat Engine will result in error.
+      To resolve error, first use `terraform refresh` command to correct state,
+      then revert the code change to ensure graceful teardown.
+    required: true
+    item_type: Api::Type::String
+  - !ruby/object:Api::Type::Enum
+    name: 'solutionType'
+    description: |
+      Solution type for Chat Engine is SOLUTION_TYPE_CHAT
+    values:
+      - :SOLUTION_TYPE_CHAT
+    default_value: :SOLUTION_TYPE_CHAT
+    immutable: true
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    description: |
+      Timestamp the Engine was created at.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    description: |
+      Timestamp the Engine was last updated.
+    output: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'chatEngineConfig'
+    description: |
+      Configurations for a chat Engine.
+    required: true
+    immutable: true
+    ignore_read: true
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'agentCreationConfig'
+        description: |
+          Configurations for a chat Engine.
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'business'
+            description: |
+              Business Name
+            required: true
+          - !ruby/object:Api::Type::String
+            name: 'defaultLanguageCode'
+            description: |
+              Default Language Code
+            required: true
+          - !ruby/object:Api::Type::String
+            name: 'timeZone'
+            description: |
+              time zone
+            required: true
+          - !ruby/object:Api::Type::String
+            name: 'location'
+            description: |
+              location
+            required: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'commonConfig'
+    description: |
+      Configurations for a chat Engine.
+    required: true
+    immutable: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'companyName'
+        description: |
+          The name of the company, business or entity that is associated with the engine
+        required: true
+        immutable: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'chatEngineMetadata'
+    description: |
+      Additional information of the Chat Engine.
+    output: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'dialogflowAgent'
```

Comment:
How does a user configure this, since it's output-only? If it's implicitly created, how does a user ensure it gets deleted when the resource is deleted? Note that implicit creations generally don't work very well in Terraform and can require making breaking changes in future releases. For example, see https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/version_5_upgrade#google_firebase_project_location-is-now-removed.

Reply:
Okay - that is still an implicit creation, which means that users won't have as much control as if they set up an explicit [`google_dialogflow_agent`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dialogflow_agent) (or is it a [`google_dialogflow_cx_agent`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dialogflow_cx_agent)?) but that's more of something to consider for the future rather than a blocker for this PR.

Will the agent automatically get cleaned up if this resource is deleted?

Reply:
@melinath  Current API behavior is that upon deletion of the chat engine, it will delete the dialogflow agent with it (no matter if the dialog flow agent is created with engine or it's an existing agent linked, there's a b/ on this to separate the lifecycle management of the two). Due to this limitation it seem like having the agent created as part of engine creation is a safer pattern as of now.

Reply:
💯 makes sense thanks!